c++/src/kj/exception.c++

```diff
@@ -1133,6 +1137,8 @@ Maybe<Exception> runCatchingExceptions(Runnable& runnable) noexcept {
   } catch (Exception& e) {
     e.truncateCommonTrace();
     return kj::mv(e);
+  } catch (CanceledException) {
```

Comment:
We should have a test case that verifies that `runCatchingExceptions()` doesn't catch `CanceledException`.

Or alternatively you could change the nesting in `FiberBase::run()` so that the `try/catch` is outside the `runCatchingExceptions()`... that would prove that `CanceledException` is rethrown.